contracts/treasury/GovernorCountingOverridable.sol

```diff
@@ -0,0 +1,229 @@
+    error VoteAlreadyCast();
+
+    function __GovernorCountingOverridable_init() internal onlyInitializing {
+        __GovernorCountingOverridable_init_unchained();
+    }
+
+    function __GovernorCountingOverridable_init_unchained() internal onlyInitializing {}
+
+    /**
+     * @dev Supported vote types. Matches Governor Bravo ordering.
+     */
+    enum VoteType {
+        Against,
+        For,
+        Abstain
+    }
+
+    /**
+     * @dev Tracks state of specicic voters in a single proposal.
+     */
+    struct ProposalVoterState {
+        bool hasVoted;
+        VoteType support;
+        // This vote deductions state is only necessary to support the case where a delegator might vote before their
+        // transcoder. When that happens, we need to deduct the delegator(s) votes before tallying the transcoder vote.
+        uint256 deductions;
+    }
+
+    /**
+     * @dev Tracks the tallying state for a proposal vote counting logic.
+     */
+    struct ProposalTally {
+        uint256 againstVotes;
+        uint256 forVotes;
+        uint256 abstainVotes;
+        mapping(address => ProposalVoterState) voters;
+    }
+
+    mapping(uint256 => ProposalTally) private _proposalTallies;
+
+    /**
+     * @dev See {IGovernor-COUNTING_MODE}.
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function COUNTING_MODE() public pure virtual override returns (string memory) {
+        return "support=bravo&quorum=for,abstain,against";
+    }
+
+    /**
+     * @dev See {IGovernor-hasVoted}.
+     */
+    function hasVoted(uint256 _proposalId, address _account) public view virtual override returns (bool) {
+        return _proposalTallies[_proposalId].voters[_account].hasVoted;
+    }
+
+    /**
+     * @dev Accessor to the internal vote counts.
+     */
+    function proposalVotes(uint256 _proposalId)
+        public
+        view
+        virtual
+        returns (
+            uint256 againstVotes,
+            uint256 forVotes,
+            uint256 abstainVotes
+        )
+    {
+        ProposalTally storage tally = _proposalTallies[_proposalId];
+        return (tally.againstVotes, tally.forVotes, tally.abstainVotes);
+    }
+
+    /**
+     * @dev See {Governor-_quorumReached}.
+     */
+    function _quorumReached(uint256 _proposalId) internal view virtual override returns (bool) {
+        (uint256 againstVotes, uint256 forVotes, uint256 abstainVotes) = proposalVotes(_proposalId);
+
+        uint256 totalVotes = againstVotes.add(forVotes).add(abstainVotes);
+
+        return totalVotes >= quorum(proposalSnapshot(_proposalId));
+    }
+
+    /**
+     * @dev See {Governor-_voteSucceeded}. In this module, the forVotes must be at least QUOTA of the total votes.
+     */
+    function _voteSucceeded(uint256 _proposalId) internal view virtual override returns (bool) {
+        (uint256 againstVotes, uint256 forVotes, ) = proposalVotes(_proposalId);
+
+        // we ignore abstain votes for vote succeeded calculation
```

Comment:
Why? If Abstain is a valid vote option shouldn't the total valid votes include abstain votes?

Reply:
I think this is mostly a naming issue on the "valid votes" variable. The logic here is that we are fetching the amount of votes with an opinion about the proposal outcome (which `abstain` doesn't [1]), and then check if we have more `for` votes than the quota of that total. I have renamed the variable to `opinionatedVotes` which I think makes more sense. This is basically the same implementation as `GovernorCountingSimple`, the only difference being that we allow the quota to be configurable instead of forcing 50% (they just do `for > against`)

Lmk if you have any other suggestion!

[1] IMO "abstain" means that you're casting a vote to communicate that you are ok with any outcome of the proposal, so you don't really have an opinion about it. That is my reasoning to think it makes sense to consider it for `_quorumReached`, but not for `_voteSucceeded`. If we did include them here there wouldn't be a behavioral difference between `abstain` and `against` (unless we removed `against` from quorum count I guess).